saleor/warehouse/management.py

```diff
@@ -198,17 +205,32 @@ def increase_stock(
     function increase `quantity_allocated`. If allocation does not exist function
     create a new allocation for this order line in this stock.
     """
+    plugins_manager = get_plugins_manager()
     stock = (
         Stock.objects.select_for_update(of=("self",))
         .filter(warehouse=warehouse, product_variant=order_line.variant)
         .first()
     )
+
     if stock:
+        product_variant_stock_trigger = product_variant_webhook_container(
+            order_line.variant,
+            warehouse.shipping_zones.values_list("channels__slug", flat=True),
+            stock.quantity,
+        )
         stock.increase_stock(quantity, commit=True)
+
     else:
+        product_variant_stock_trigger = product_variant_webhook_container(
+            order_line.variant,
+            warehouse.shipping_zones.values_list("channels__slug", flat=True),
+            prev_quantity=0,
+        )
+
         stock = Stock.objects.create(
             warehouse=warehouse, product_variant=order_line.variant, quantity=quantity
         )
+
     if allocate:
```

Comment:
Maybe I am missing something, but Shouldn't we take into consideration that stock can be imedietly allocated here? So calling `back_in_stock` will not be true?
Related comment: https://github.com/mirumee/saleor/pull/7590/files#r675384459

Reply:
But allocation itself will not increase or decrease the total quantity of a given Stock object.  Sending webhooks after each allocation changing (for example, via orderLineCRUD mutations) can lead us to unwanted side effects.

Reply:
Maybe I am wrong but re-creating stock and then immediately create allocation means that the new stock which was just created is already allocated, which means that you don't have any free stock.